src/AmplNLWriter.jl

```diff
@@ -316,10 +183,16 @@ MOI.get(model::Optimizer, ::MOI.SolverName) = "AmplNLWriter"
 MOI.supports(::Optimizer, ::MOI.NLPBlock) = true
 
 MOI.supports(::Optimizer, ::MOI.Name) = true
```

Comment:
Seeing this code makes me wonder if it could be useful to have an abstract type of model in MOI (say, `AbstractDecoratorModel`) that mostly pass calls to the submodel (`model.inner`), except if overridden (like here). I think it's an implementation of the decorator design pattern. It could save similar code for models that just wrap another model.

Reply:
More precisely, it would be about factoring code like this: 

```julia
function MOI.supports(model::Optimizer, attr::MOI.AnyAttribute, x...) 
    return MOI.supports(model.inner, attr, x...)::Bool
end

function MOI.get(model::Optimizer, attr::MOI.AnyAttribute, x...) 
    return MOI.get(model.inner, attr, x...)
end

function MOI.set(model::Optimizer, attr::MOI.AnyAttribute, x...) 
    MOI.set(model.inner, attr, x...)
    return
end
```

(Still untested.)

Reply:
There's an issue for that: https://github.com/jump-dev/MathOptInterface.jl/issues/1399

Reply:
I thought this issue was more general than just forwarding…

Reply:
It's the same issue. We want to have some tooling to automatically forward all the methods (or a subset of them) without ambiguity errors.